plugins/CoreAdminHome/angularjs/trackingcode/jstrackingcode.controller.js

```diff
@@ -66,12 +67,23 @@
                 filter_limit: '-1'
             }).then(function (data) {
                 self.siteUrls[idSite] = data || [];
-
-                // re-enable controls
                 self.isLoading = false;
+                callback();
+            });
 
+            self.isLoading = true;
+            // Load site excludedQueryParams
+            piwikApi.fetch({
+                module: 'API',
+                method: 'Overlay.getExcludedQueryParameters',
+                idSite: idSite,
+                filter_limit: '-1'
+            }).then(function (data) {
+                self.siteExcludedQueryParams[idSite] = data || [];
+                self.isLoading = false;
                 callback();
             });
+
```

Comment:
This would actually call `callback` twice. But it should be loaded when everything is finished, right? So guess the simplest solution would be to execute the queries not in parallel by moving the second request into the first one's `then` part, like:

```suggestion
                // Load site excludedQueryParams
                piwikApi.fetch({
                    module: 'API',
                    method: 'Overlay.getExcludedQueryParameters',
                    idSite: idSite,
                    filter_limit: '-1'
                }).then(function (data) {
                    self.siteExcludedQueryParams[idSite] = data || [];
                    self.isLoading = false;
                    callback();
                });
            });

```

otherwise the `isLoading` wouldn't be set correctly all the time, as one request might still be running...